map/Region.dhall

```diff
@@ -0,0 +1,269 @@
+                        , ap-northeast-1 = generate Region.ap-northeast-1
+                        , ap-northeast-2 = generate Region.ap-northeast-2
+                        , ap-northeast-3 = generate Region.ap-northeast-3
+                        , ap-south-1 = generate Region.ap-south-1
+                        , ap-southeast-1 = generate Region.ap-southeast-1
+                        , ap-southeast-2 = generate Region.ap-southeast-2
+                        , ca-central-1 = generate Region.ca-central-1
+                        , cn-north-1 = generate Region.cn-north-1
+                        , cn-northwest-1 = generate Region.cn-northwest-1
+                        , eu-central-1 = generate Region.eu-central-1
+                        , eu-north-1 = generate Region.eu-north-1
+                        , eu-south-1 = generate Region.eu-south-1
+                        , eu-west-1 = generate Region.eu-west-1
+                        , eu-west-2 = generate Region.eu-west-2
+                        , eu-west-3 = generate Region.eu-west-3
+                        , me-south-1 = generate Region.me-south-1
+                        , sa-east-1 = generate Region.sa-east-1
+                        , us-gov-east-1 = generate Region.us-gov-east-1
+                        , us-gov-west-1 = generate Region.us-gov-west-1
+                        , us-east-1 = generate Region.us-east-1
+                        , us-east-2 = generate Region.us-east-2
+                        , us-west-1 = generate Region.us-west-1
+                        , us-west-2 = generate Region.us-west-2
+                        }
+                        region
+
+          in  { function = render, Params, letters }
+      , partition =
+          let render
+              : Region → Text
+              = λ(region : Region) →
+                  merge
+                    { af-south-1 = "aws"
+                    , ap-east-1 = "aws"
+                    , ap-northeast-1 = "aws"
+                    , ap-northeast-2 = "aws"
+                    , ap-northeast-3 = "aws"
+                    , ap-south-1 = "aws"
+                    , ap-southeast-1 = "aws"
+                    , ap-southeast-2 = "aws"
+                    , ca-central-1 = "aws"
+                    , cn-north-1 = "aws-cn"
+                    , cn-northwest-1 = "aws-cn"
+                    , eu-central-1 = "aws"
+                    , eu-north-1 = "aws"
+                    , eu-south-1 = "aws"
+                    , eu-west-1 = "aws"
+                    , eu-west-2 = "aws"
+                    , eu-west-3 = "aws"
+                    , me-south-1 = "aws"
+                    , sa-east-1 = "aws"
+                    , us-gov-east-1 = "aws-us-gov"
+                    , us-gov-west-1 = "aws-us-gov"
+                    , us-east-1 = "aws"
+                    , us-east-2 = "aws"
+                    , us-west-1 = "aws"
+                    , us-west-2 = "aws"
+                    }
+                    region
```

Comment:
What is the purpose of that? Any other option other then `aws`?

Reply:
https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html

this is for building `arn` , and its different for `cn & us-gov`